BuilderDesignPattern/src/main/java/com/designPattern2/House.java

```diff
@@ -0,0 +1,39 @@
+package com.designPattern2;
+
+public class House {

```

Comment:
Do you think the `House` and `Flat` classes should share a common interface?

Reply:
In this instance maybe however to align with the typical Builder pattern if we wanted to have different methods in each then splitting them would be the correct thing to do.